decisions/0001-file-sharing.md

```diff
@@ -0,0 +1,103 @@
+- The backend will provide to the preprocessing pipeline pre-signed URLs to read the original files. If the pipeline would like to upload processed files, it calls the same `/files/request-uploads` endpoint.
+- During the review, the user receives pre-signed URLs to download the original and processed files.
+
+### S3 and (semi-)public write-only link
+
+- Authenticated users can see on the website information to upload to files to a shared S3 bucket.
+- They should choose a random object names to avoid conflicts with uploads of other users.
+- They state the object name in the metadata file when they submit (or revise).
+
+### User-controlled S3
+
+- Users upload files to their own S3 bucket and give Loculus read access.
+
+
+### Addon: File hashing
+
+Every uploaded file will be hashed. If a file has been already uploaded before, it will not be uploaded again. Maybe, it is possible to get a hash from S3 directly (see https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html) but it has not been clarified whether this is widely supported by different S3 providers/software.
+
+This can be combined with any of the other options.
+
+## Decision Outcome
+
+Chosen option: "{title of option 1}", because {justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force {force} | … | comes out best (see below)}.
+
+<!-- This is an optional element. Feel free to remove. -->
+### Consequences
+
+* Good, because {positive consequence, e.g., improvement of one or more desired qualities, …}
+* Bad, because {negative consequence, e.g., compromising one or more desired qualities, …}
+* … <!-- numbers of consequences can vary -->
+
+<!-- This is an optional element. Feel free to remove. -->
+### Confirmation
+
+{Describe how the implementation / compliance of the ADR can/will be confirmed. Is there any automated or manual fitness function? If so, list it and explain how it is applied. Is the chosen design and its implementation in line with the decision? E.g., a design/code review or a test with a library such as ArchUnit can help validate this. Note that although we classify this element as optional, it is included in many ADRs.}
+
+<!-- This is an optional element. Feel free to remove. -->
+## Pros and Cons of the Options
+
+### S3 and pre-signed URLs
+
+* Good, because this gives the backend fine-grained control over the uploaded files:
+  * It controls where files are uploaded to, avoiding users overwriting each other with certainty.
+  * It can set a max file size for each file.
+  * It can limit number of uploads (in total/within a time period/for a user/group/etc.) to limit the use of resources and potential for abuse
+* Bad, because (compared to option "S3 and (semi-)public write-only link"), users need to call an additional endpoint to receive the pre-signed URLs.
+
```

Comment:
Additionally users need to update the metadata with the links of each raw read that they have uploaded (additionally your proposal mentioned an additional call to say submission has been completed - so that is one more step)